app/models/availability.rb

```diff
@@ -39,4 +41,15 @@ def self.process_data
         { 'Date' => date, 'Count' => count }
       end
   end
+
+  private
+
+  def cancel_duplicates
+    previous_availabilities = person.availabilities.for_time_span(start_time..end_time).order(:created_at)
+    previous_availabilities.each do |a|
+      if start_time == a.start_time && end_time == a.end_time
+        a.update(status: "Cancelled") unless self == a
```

Comment:
I agree with what @pcai has said about rewriting the block. If we want to use `for_time_span`, I think it'd look something like this:

```ruby
person.availabilities.for_time_span(start_time..end_time).where.not(id: id).update_all(status: "Cancelled")
```